src/algebra/char_p/exp_char.lean

```diff
@@ -0,0 +1,113 @@
+/-
+Copyright (c) 2021 Jakob Scholbach. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Jakob Scholbach.
+-/
+import algebra.char_p.basic
+import algebra.char_zero
+import data.nat.prime
+
+/-!
+# Exponential characteristic
+
+This file defines the exponential characteristic of an integral domain,
+and establishes a few basic facts relating the exponential characteristic to the (ordinary)
+characteristic.
+
+## Main results
+- `exp_char`: the definition of exponential characteristic
+- `exp_char_is_prime_or_one`: the exponential characteristic is a prime or one
+- `char_eq_exp_char_iff`: the characteristic equals the exponential characteristic iff the
+  characteristic is prime
+
+## Tags
+exponential characteristic, characteristic
+-/
+
+universe u
+variables (R : Type u) [integral_domain R]
+
+/-- The definition of the exponential characteristic of a semiring. -/
+class exp_char (R : Type u) [semiring R] (q : ℕ) : Prop :=
+(exp_char_def : (q = 1 ∧ char_zero R) ∨ (q.prime ∧ char_p R q))
+
+/-- The exponential characteristic is one if the characteristic is zero. -/
+lemma exp_char_one_of_char_zero (q : ℕ) [hp : char_p R 0] [hq : exp_char R q] :
+q = 1 :=
+begin
+  cases hq.exp_char_def with q_one q_prime,
+  { exact q_one.1, },
+  { exact false.elim (lt_irrefl _ ((hp.eq R q_prime.2).symm ▸ q_prime.1 : (0 : ℕ).prime).pos) }
+end
+
+/-- The exponential characteristic is one if the characteristic is zero. -/
+lemma char_zero_of_exp_char_one (p : ℕ) [hp : char_p R p] [hq : exp_char R 1] :
+p = 0 :=
+begin
+  cases hq.exp_char_def with q_one q_prime,
+  { haveI := q_one.2,
+    exact char_p.eq R hp infer_instance, },
+  { haveI := q_prime.2,
+    exact false.elim (char_p.char_ne_one R 1 rfl), }
+end
+
+/-- The exponential characteristic is one if the characteristic is zero. -/
+instance char_zero_of_exp_char_one' [hq : exp_char R 1] : char_zero R :=
```

Comment:
```suggestion
@[priority 100] -- see Note [lower instance priority]
instance char_zero_of_exp_char_one' [hq : exp_char R 1] : char_zero R :=
```
This will make the linter happy

Reply:
Or at least happier: the linter always seems stressed...

Reply:
OK, done.